pkg/dependencies/factory.go

```diff
@@ -162,11 +162,25 @@ func (f *Factory) WithRegistryMirror(mirror string) *Factory {
 	return f
 }
 
-func (f *Factory) WithProxyConfiguration(proxyConfig map[string]string) *Factory {
+func (f *Factory) UseProxyConfiguration(proxyConfig map[string]string) *Factory {
 	f.proxyConfiguration = proxyConfig
 	return f
 }
 
+func (f *Factory) WithProxyConfiguration() *Factory {
```

Comment:
It should be already tested by I would add this explicitly to a test and assert not-nil